e2e_playwright/websocket_reconnects_test.py

```diff
@@ -30,87 +31,11 @@
-    """
-
-    return app.evaluate(
-        """async ([expectedStatus]) => {
-                // the first call to resolve will be the one returned to the caller
-                // so its either the observed status or the timeout. Subsequent
-                // calls are no-ops.
-                const p = new Promise((resolve) => {
-                    // Define a timeoutId so that we can cancel the timeout in the
-                    // callback upon success
-                    let timeoutId = null
-                    let resolved = false
-                    const callback = (mutationList, observer) => {
-                        if (!mutationList || mutationList.length === 0) {
-                            return
-                        }
-                        const target = mutationList[0].target
-                        if (!target) {
-                            return
-                        }
-                        let state = target
-                                        .getAttribute('data-test-connection-state')
-                                        .toUpperCase();
-                        if (state.indexOf(expectedStatus.toUpperCase()) > -1) {
-                            resolved = true
-                            if (timeoutId) clearTimeout(timeoutId)
-                            if (observer) observer.disconnect()
-                            resolve(state)
-                        }
-                    }
-                    const observer = new MutationObserver(callback);
-                    // Observe app status for changes
-                    const targetNode = document.querySelector('[data-testid=stApp]')
-                    if (!targetNode) {
-                        resolve("stApp not found")
-                        return
-                    }
-                    const config = {
-                        childList: false,
-                        subtree: false,
-                        attributeFilter: ['data-test-connection-state']
-                    };
-                    observer.observe(targetNode, config);
-
-            """
-        + callable_action
-        + """
-                    if (!resolved) {
-                        timeoutId = setTimeout(() => {
-                            if (observer) observer.disconnect()
-                            resolve(`timeout: did not observe status '${expectedStatus}'`)
-                            return
-                        }, 1500);
-                    }
-                })
-
-                const status = await p
-                return status
-            }
-            """,
-        [expected_status],
-    )
-
-
 def test_dont_observe_invalid_status(
     app: Page,
 ):
     """Test that unknown status is not observed and raises an error."""
-    status = _get_status(app, "Connecting2", DISCONNECT_WEBSOCKET_ACTION)
+    status = get_connection_status(app, "Connecting2", DISCONNECT_WEBSOCKET_ACTION)
```

Comment:
Wouldn't it be possible to make it an `expect_connection_status` method that doesn't return anything?

Reply:
I think that's a good idea 👍